staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -448,3 +449,57 @@ func TestPanicPropagated(t *testing.T) {
 		t.Errorf("timeout: the panic failed to propagate from the controller run method!")
 	}
 }
+
+// TestControllerRunWithStopOptions confirms that a controller stops itself
+// when the underlying reflector hits a list error by starting a controller
+// and in a separatre goroutine and waiting for the controller stops itself
+// before cancelling the context. If the context is stopped before the test
+// timeout, that means the controller stopped itself as expected.
+func TestControllerRunWithStopOptions(t *testing.T) {
+	// waitTime is how long to wait for the stopOnError=false case to pass.
+	// It needs to just be long enough for the underlying reflector to make
+	// its initial list call and error out.
+	//
+	// We begin seeing flakes every couple thousand runs at 10ms, so 100ms
+	// was chosen as an order of magnitude higher and does not seem to ever flake.
```

Comment:
is this true even when running under [stress](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md#deflaking-unit-tests)? CI CPU is significantly lower than dev machines

Reply:
yup, stress is what I used to test this, but see below, happy to bump it up if it's a concern.